packages/rollup-plugin/src/rollupUtils.ts

```diff
@@ -6,30 +6,53 @@ import type { SourcemapInfo } from './types'
 /**
  * Extracts the data we need for sourcemap upload from the bundle
  */
-export function extractSourcemapDataFromBundle ( 
-  outputOptions: NormalizedOutputOptions, 
-  bundle: OutputBundle
+export function extractSourcemapDataFromBundle (
+  outputOptions: NormalizedOutputOptions,
+  bundle: OutputBundle,
+  ignorePaths: Array<string | RegExp> = []
 ): SourcemapInfo[] {
   const sourceMaps = Object.values(bundle).filter(isSourcemap)
-  
+
   return sourceMaps.map(sourcemap => {
     return formatSourcemapData(outputOptions, sourcemap)
-  })
+  }).filter((sourcemap) => isNotIgnored(sourcemap, ignorePaths))
 }
 
 function isSourcemap(file: OutputAsset | OutputChunk): file is OutputAsset {
-  return file.type === 'asset' && file.fileName.endsWith('.js.map')
+  if (file.type !== 'asset' || !file.fileName.endsWith('.js.map') || !file.source) {
+    return false
+  }
+
+  const source = typeof file.source === 'string' ? file.source : file.source.toString()
+  const json = JSON.parse(source)
+
+  return json.sourcesContent && json.sourcesContent.length > 0
```

Comment:
Since this function should return a boolean, you may want to coerce json.sourcesContent to be a boolean (eg `!!json.sourcesContent`).